libsolidity/interface/CompilerStack.cpp

```diff
@@ -516,7 +516,17 @@ bool CompilerStack::compile()
 			if (auto contract = dynamic_cast<ContractDefinition const*>(node.get()))
 				if (isRequestedContract(*contract))
 				{
-					compileContract(*contract, otherCompilers);
+					try
+					{
+						compileContract(*contract, otherCompilers);
+					}
+					catch (Error const& _error)
+					{
+						if (_error.type() != Error::Type::CodeGenerationError)
+							throw;
```

Comment:
We have a couple of top-level `catch`s for `Errors`, so I re-throw to keep the old behavior if such exception happens.

(I am not sure if/where we throw other `Error`s though. Maybe we should review that.)